File: 111.MinimumDepthofBinaryTree/clywin123/minDepth.py
class Solution:
    # @param {TreeNode} root
    # @return {integer}
    def minDepth(self, root):
        if(not root):
            return 0
        def _minDepth(root,value):
            if(not root):
                return float("inf")
            if(root.left==None and root.right==None):
                value += 1
                return value
            value+=1
            le = _minDepth(root.left,value)
            ri = _minDepth(root.right,value)
            return min(le,ri)
        return _minDepth(root,0)

File: 111.MinimumDepthofBinaryTree/wyrj/minDepth.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (!root) return 0;
    let depth = 0;
    let q = [root];
    let nq = [];
    while (q.length > 0) {
        depth++;
        for (let i = 0; i < q.length; i++) {
            if (!q[i].left && !q[i].right) return depth;
            if (q[i].left) nq.push(q[i].left);
            if (q[i].right) nq.push(q[i].right);
        }
        q = nq;
        nq = [];
    }
    return depth;
};
